### DISCUSSION TITLE
Integrate real time debugger issue(Integrate Segger RTT viewer with STM32F411RE in eclipse ide)

### DISCUSSION THREAD

**Discussion by Sami2098** [eclipse-embed-cdt/eclipse-plugins#613] [Debug Plug-ins Q&A]
hello all,
I was trying to integrate Segger RTT viewer with STM32F411RE in eclipse ide
all time when I connect RTT viewer this error occure in RTT viewer

"ERROR: : Terminal change character (IAC) followed by invalid character (non-hexadecimal)."

I also attach debugger configuration

![Debug_config](https://github.com/user-attachments/assets/5a24aaa2-46e7-4b31-aa71-4d84a32d95d8)

**Comment by ilg-ul** [comment]
I have no experience with Segger RTT viewer connecting to OpenOCD.

I suggest you first experiment with a debug session started by hand in a terminal and only if you succeed consider using it in Eclipse.

**Comment by TommyMurphyTM1234** [comment]
In case it helps at all, that error message crops up on the Segger forums and in other places in relation to the Segger RTT viewer. You can find these by using a web search. As far as I can tell the message occurs due to the nature of the RTT data being sent from the target so you might want to check/clarify what exactly you're sending from the target side.
